### PR TITLE
Refactor tasks into decorators

### LINKED ISSUE #3973: Refactor Task subclasses into decorated functions
Breaking out the task steps in order to reduce shared state on the tasks means
that we no longer need custom classes here. We should reduce the Task subclasses
back down to decorated functions, which preserve the naming, queue, and other
properties of the task.

- [ ] project.tasks
- [x] core.util.tasks

### PR DESCRIPTION
**Pull request by stsewd**
Closes #3973

This will also helps with https://github.com/rtfd/readthedocs.org/issues/3984 later

### CODE REVIEW

File: readthedocs/projects/tasks.py [hunk 1]
```diff
@@ -239,17 +230,10 @@ def run(self, version_pk):  # pylint: disable=arguments-differ
...
-
-    max_retries = 5
-    default_retry_delay = (7 * 60)
-    name = __name__ + '.update_docs'
-
-    def run(self, *args, **kwargs):
-        step = UpdateDocsTaskStep(task=self)
-        return step.run(*args, **kwargs)
+@app.task(bind=True, max_retries=5, default_retry_delay=7 * 60)
+def update_docs_task(self, project_id, *args, **kwargs):
```
Comment (stsewd) [inline readthedocs/projects/tasks.py]: Pylint doesn't like binded task, the self parameter gets injected by the decorator.

File: readthedocs/core/management/commands/pull.py [hunk 2]
```diff
@@ -20,6 +20,4 @@ def handle(self, *args, **options):
         if args:
             for slug in args:
                 version = utils.version_from_slug(slug, LATEST)
-                tasks.SyncRepositoryTask().run(
-                    version.pk,
-                )
+                tasks.sync_repository_task(version.pk)
```
Comment (stsewd) [inline readthedocs/core/management/commands/pull.py]: I assume that calling directly to run is the same as executing the task on the same host.

**Comment by stsewd** [comment]
I think this is ready, we can refactor more: classes aren't needed here, we can refactor those to be just functions. Let me know if you want me to do that here, or maybe we can do it in another PR.

Also, I tested with some builds locally (also some commands), it works.

File: readthedocs/core/management/commands/pull.py (hunk 2 above)
Comment (agjohnson) [reply readthedocs/core/management/commands/pull.py]: Yeah, this should be equal

**Comment by stsewd** [comment]
Looks like need to fix some linter errors

**Comment by agjohnson** [comment]
Yeah i think in part due to my resolve. Seems to be okay now though

**Comment by stsewd** [comment]
Error fixed, it was because of https://github.com/rtfd/readthedocs.org/pull/4653

**Pull request merged**